### ISSUE TITLE
Emoji vs Emoji_Presentation

### ISSUE THREAD

**Issue by brodieG** [krlmlr/utf8#9]
AFAICT in determining whether a code-point should be wide or not the code (or at least the code that gens the tables) relies on checking whether it's [emoji or emoji_presentation ](https://github.com/patperry/utf8lite/blob/master/util/gen-charwidth.py#L44) (not 100% certain):

```
# https://www.unicode.org/reports/tr51/#def_basic_emoji_set
emoji = ((emoji_props['Emoji'] - emoji_props['Emoji_Component'])
         | emoji_props['Emoji_Presentation'])
```
But the current tr51 states:

> The emoji code points are those with property values Emoji=Yes, Emoji_Component=No, and Emoji_Presentation=Yes.

It seems the `|` above is effectively doing an `or`, though I do not know python so I have no idea if the code is doing what I think it's doing.  However, I do see:

```
utf8::utf8_width(c('\u2139', '\u2728'))
## [1] 2 2
```
u2139 is not in the "Emoji Presentation" section of [Emoji_data](https://unicode.org/Public/emoji/12.1/emoji-data.txt), but u2728 is.

On my system (mojave OS X terminal) this is what I see:

![image](https://user-images.githubusercontent.com/6105908/75839938-034cd300-5d98-11ea-8af1-9bccfd669832.png)

I don't pretend that my system is the end all be-all in terms of the correct display computation, but it appears to behave as per tr51.

There is additional ambiguity with some emoji with text presentation that actually have a wide-ish text presentation:

![image](https://user-images.githubusercontent.com/6105908/75839979-2d05fa00-5d98-11ea-8748-c3343cac96ab.png)

Though clearly the terminal treats it as 1-wide (FWIW, until recently the terminal also treated normal emojis as 1-wide...).

**Comment by krlmlr** [comment]
Thanks. It looks like the `|` should be an `&` indeed. Would you like to recreate the tables and test if this fixes `utf8_width()` ?

**Comment by brodieG** [comment]
I'll give it a shot, but again probably will take me a week or a bit more to get to it.  If you are in a  rush for release feel free to do it and let me know if you do (so we don't end up both doing it).

**Comment by patperry** [comment]
Agree that the text seems to suggest `&`. I experimented with this, noted that U+1F1E6 (regional indicator 'A') changes from width 2 to 1, which seems correct.

**Comment by patperry** [comment]
I just switched "&" to "|" in the `emoji-width` branch of `utf8lite`: https://github.com/patperry/utf8lite/tree/emoji-width . This fixes the cases @brodieG mentioned. 

I haven't done a detailed before/after comparison, other than spot-checking a few other examples that changed.

**Comment by brodieG** [comment]
I did not do a before / after, but I did compare the full set against R4.0.5 (also against R4.1, but that highlighted 4.1 has some regressions).  On the whole things seem reasonable.  Because R4.0.5 is on 12.1.0 there are differences, but they seem either related to the UCD change, or to different assumptions (e.g. unspecified code point widths).

@krlmlr presumably this will require updating the `utf8lite` sources embedded in `utf8`.
